src/Type/Definition/FieldArgument.php

```diff
@@ -28,14 +28,17 @@ class FieldArgument
     /** @var mixed[] */
     public $config;
 
-    /** @var InputType */
+    /** @var InputType&Type */
     private $type;
 
-    /** @var bool */
-    private $defaultValueExists = false;
```

Comment:
I removed this because you can do the same logic by checking the config bucket, but there was and still is a potential bug, here--`defaultValue` is public with no setter, so if no default value is passed at construction time but is set after the fact, then `defaultValueExists` will still return `false`.

Reply:
I don't intend to fight this battle in this PR, but there _are_ ways to have read-only properties. For example:

```
class FieldArgument {
   private $_defaultValue;

   public function __get($name) {
       switch($name) {
          case 'defaultValue`:
             return $_defaultValue;
       } 
   }
}
```